source/particle/interpolator/cell_average.cc

```diff
@@ -56,29 +56,60 @@ namespace aspect
         else
           found_cell = cell;
 
-        const types::LevelInd cell_index = std::make_pair<unsigned int, unsigned int> (found_cell->level(),found_cell->index());
+        const types::LevelInd cell_index = std::make_pair(found_cell->level(),found_cell->index());
         const std::pair<typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator,
               typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator> particle_range = particles.equal_range(cell_index);
 
         const unsigned int n_particles = std::distance(particle_range.first,particle_range.second);
         const unsigned int n_properties = particles.begin()->second.get_properties().size();
         std::vector<double> cell_properties (n_properties,0.0);
 
-        AssertThrow(n_particles != 0,
-                    ExcMessage("At least one cell contained no particles. The 'cell "
-                               "average' interpolation scheme does not support this case. "));
-
-        for (typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator particle = particle_range.first;
-             particle != particle_range.second; ++particle)
+        if (n_particles > 0)
           {
-            const std::vector<double> &particle_properties = particle->second.get_properties();
+            for (typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator particle = particle_range.first;
+                 particle != particle_range.second; ++particle)
+              {
+                const std::vector<double> &particle_properties = particle->second.get_properties();
+
+                for (unsigned int i = 0; i < n_properties; ++i)
+                  cell_properties[i] += particle_properties[i];
```

Comment:
This is too late now, but I wonder if we could have made our life easier by using `std::valarray` instead of `std::vector` for particle properties, since it has a built-in `operator+` for the whole collection of elements. This would have allowed to write these three lines into one.